setup.sh

```diff
@@ -84,601 +120,224 @@ vercomp () {
+    $KUZZLE_PUSH_ANALYTICS'{"type": "docker-version-mismatch", "uid": "'$ANALYTICS_UUID'", "os": "'$OS'"}' $ANALYTICS_URL &> /dev/null    
+    exit $DOCKER_VERSION_MISMATCH
+  fi
 
-setupMapCount() {
-  SYSCTL_CONF_FILE=/etc/sysctl.conf
-  writeBold "    This script can set it automatically or you can do it manually."
-  write     "    More information at https://www.elastic.co/guide/en/elasticsearch/reference/5.x/vm-max-map-count.html"
-  while [[ "$setVmParam" != [yYnN] ]]
-  do
-    echo
-    writeBold "[❓] Do you want to set the vm.max_map_count now? (y/N) "
-    echo -n "> "
-    read setVmParam trash
-    case "$setVmParam" in
-      [yY])
-        echo
-        writeBold "Setting kernel variable vm.max_map_count to $REQUIRED_MAP_COUNT..."
-        sysctl -w vm.max_map_count=$REQUIRED_MAP_COUNT
-        if [ -z "$MAP_COUNT" ]; then
-          echo "vm.max_map_count=$REQUIRED_MAP_COUNT" >> $SYSCTL_CONF_FILE
-        else
-          sed 's/vm.max_map_count=.+/vm.max_map_count=$REQUIRED_MAP_COUNT/g' > ${TMPDIR-/tmp}/sysctl.tmp
-          mv ${TMPDIR-/tmp}/sysctl.tmp $SYSCTL_CONF_FILE
-        fi
-        writeBold "$GREEN" "[✔] Kernel variable successfully set."
-        ;;
-      [nN] | '')
-        echo
-        writeBold "$BLUE" "Ok. Please set the kernel variable and re-run this script."
-        echo
-        exit 0
-        ;;
-      *)
-      echo
-      writeBold "$RED" "[✖] Please, answer Y or N."
-      ;;
-    esac
-  done
+  # Check of vm.max_map_count is at least $MIN_MAX_MAP_COUNT
+  VM_MAX_MAP_COUNT=$(sysctl -n vm.max_map_count)
+  if [ -z "${VM_MAX_MAP_COUNT}" ] || [ ${VM_MAX_MAP_COUNT} -lt $MIN_MAX_MAP_COUNT ]; then
+    >&2 echo
+    >&2 echo $RED"The current value of the kernel configuration variable vm.max_map_count (${VM_MAX_MAP_COUNT})"
+    >&2 echo "is lower than the required one ($MIN_MAX_MAP_COUNT+)."
+    >&2 echo "In order to make ElasticSearch working please set it by using on root: (more at https://www.elastic.co/guide/en/elasticsearch/reference/5.x/vm-max-map-count.html)"
+    >&2 echo $BLUE$BOLD"sysctl -w vm.max_map_count=$MIN_MAX_MAP_COUNT"
+    >&2 echo $RED"If you want to persist it please edit the $BLUE$BOLD/etc/sysctl.conf$NORMAL$RED file"
+    >&2 echo "and add $BLUE$BOLD vm.max_map_count=$MIN_MAX_MAP_COUNT$NORMAL$RED in it."$NORMAL
+    $KUZZLE_PUSH_ANALYTICS'{"type": "wrong-max_map_count", "uid": "'$ANALYTICS_UUID'", "os": "'$OS'"}' $ANALYTICS_URL &> /dev/null    
+    exit $MIN_MAX_MAP_COUNT_MISMATCH
+  fi
 }
 
-collectPersonalData() {
+download_docker_compose_yml() {
+  local RETRY=0
+
+  echo 
+  echo $BLUE"Downloading Kuzzle docker-compose.yml file..."$NORMAL
+
+  TEST=$(eval "$KUZZLE_CHECK_DOCKER_COMPOSE_YML_HTTP_STATUS_CODE")
+  while [ $TEST -ne 200 ];
+    do
+      if [ $RETRY -gt $DOWNLOAD_DOCKER_COMPOSE_YML_MAX_RETRY ]; then
+        >&2 echo $RED"Cannot download $COMPOSE_YML_URL."
+        >&2 echo "Please ensure you have internet or try again later."
```

Comment:
Again, that error takes me back to year 2000, where softwares constantly asked me to check my internet connection.
If the error is the result of the unavailability of the docker-compose file (and it already happened), then that message can be infuriating, because we're basically saying: "YOU are at fault, this cannot be from our part" (that's how I understand these kind of errors anyway)
Either the curl/wget exit code states that there is no network connection and the message should reflect that, or we should provide the HTTP error code in the error message, and ask the user to contact us and provide the returned error.
That way we would be informed more quickly that there is a problem with our cdn